dev/NavigationView/NavigationView.cpp

```diff
@@ -2132,72 +2395,43 @@ void NavigationView::UpdateNavigationViewUseSystemVisual()
 {
     if (SharedHelpers::IsRS1OrHigher() && !ShouldPreserveNavigationViewRS4Behavior() && m_appliedTemplate)
     {
-        auto showFocusVisual = SelectionFollowsFocus() == winrt::NavigationViewSelectionFollowsFocus::Disabled;
-
-        PropagateChangeToNavigationViewLists(NavigationViewPropagateTarget::LeftListView,
-            [showFocusVisual](NavigationViewList* list)
-        {
-            list->SetShowFocusVisual(showFocusVisual);
-        }
-        );
+        auto showFocusVisual = ShouldShowFocusVisual();
 
-        PropagateChangeToNavigationViewLists(NavigationViewPropagateTarget::TopListView,
-            [showFocusVisual](NavigationViewList* list)
+        if (IsTopNavigationView())
         {
-            list->SetShowFocusVisual(showFocusVisual);
+            PropagateShowFocusVisualToAllNavigationViewItemsInRepeater(m_leftNavRepeater.get(), showFocusVisual);
         }
-        );
-    }
-}
-
-void NavigationView::SetNavigationViewListPosition(winrt::ListView& listView, NavigationViewListPosition position)
-{
-    if (listView)
-    {
-        if (auto navigationViewList = listView.try_as<winrt::NavigationViewList>())
+        else
```

Comment:
>else [](start = 8, length = 4)

shoudn't we hit the overflow items as well?

Reply:
No, this implementation disables focusvisuals when selectionfollowsfocus. However, if you look at the current implementation of NavigationView, selectionfollowsfocus is not a thing in the overflow menu (due to the overflow menu automatically closing upon selection, hence navigating using keyboard focus would be problematic).